lib/js-file.js

```diff
@@ -415,11 +408,11 @@ JsFile.prototype = {
      */
     getNodesByType: function(type) {
         if (typeof type === 'string') {
-            return this._index[type] || [];
+            return this._program.selectNodesByType(type);
         } else {
             var result = [];
             for (var i = 0, l = type.length; i < l; i++) {
-                var nodes = this._index[type[i]];
+                var nodes = this._program.selectNodesByType(type[i]);
```

Comment:
Looks much better, although it seems there is no need for the `for` anymore, some of the `Array#<methods>` or `lodash` methods should do the job?